src/domain/accounts/domain/src/repos/access_token_repository.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use internal_error::InternalError;
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::{AccessToken, Account};
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait AccessTokenRepository: Send + Sync {
+    async fn create_access_token(
+        &self,
+        access_token: &AccessToken,
+    ) -> Result<(), CreateAccessTokenError>;
+
+    async fn get_token_by_id(&self, token_id: &Uuid) -> Result<AccessToken, GetAccessTokenError>;
+
+    async fn get_access_tokens(
+        &self,
+        pagination: &AccessTokenPaginationOpts,
+    ) -> Result<Vec<AccessToken>, GetAccessTokenError>;
+
```

Comment:
Don't know yet where is this used, but as an API it looks dangerous.
Who will ever need access tokens traversal without filtering by account??

Reply:
I see that it is only used in tests. Still, we will need a GraphQL API to list current user tokens, so a query like this, but with account filter  - and that is good enough for your tests too.